common/djangoapps/student/views.py

```diff
@@ -874,20 +849,6 @@ def change_enrollment(request, check_access=True):
 
         # Otherwise, there is only one mode available (the default)
         return HttpResponse()
-
-    elif action == "add_to_cart":
```

Comment:
@bradenmacdonald I see that the functionality moved to the auth completion view, which makes sense given your refactor. The question now is why the shoppingcart addition isn't working for @chrisndodge, even though the new code is tested.